### PR TITLE
Create skeleton Periodic Background Sync spec.

### PR DESCRIPTION
**Pull request by mugdhalakhani**
@rayankans, @beverloo PTAL.

I have chosen to create a separate spec for Periodic Background Sync in lthe spec/ folder, imitating what we've done with the explainers/ folder. If this isn't OK, I can create two spec folders with their own index.bs|.html files.

If this is fine, I'll send follow-up PRs for the remaining sections.

### CODE REVIEW

File: spec/PeridicBackgroundSync-index.bs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+Shortname: periodic-background-sync
+Level: 1
+Editor: Mugdha Lakhani, Google, nator@chromium.org
+Abstract: This specification describes a method that enables web applications to periodically synchronize data and content in the background.
+Group: wicg
+Repository: WICG/BackgroundSync
+Link Defaults: html (dfn) allowed to show a popup/event handler idl attribute/global object/in parallel/incumbent settings object/perform a microtask checkpoint/queue a task/script execution environment
+</pre>
+
+<pre class=biblio>
```
Comment (rayankans) [inline spec/PeridicBackgroundSync-index.bs]: None of the links in lines 14->81 are being used

File: spec/PeridicBackgroundSync-index.bs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+  As this API relies on service workers, functionality provided by this API is only available in a <a>secure context</a>.
+  <div class="example">
+    Requesting a <a>periodic Background Sync opportunity</a> at a daily frequency from a <a>browsing context</a>:
+
+    <pre class="lang-js">
+      function getLatestNews() {
```
Comment (rayankans) [inline spec/PeridicBackgroundSync-index.bs]: async

File: spec/index.html [hunk 3]
```diff
@@ -1029,7 +1029,7 @@
 		}
 	/* } */
```
Comment (rayankans) [inline spec/index.html]: can you revert the changes to this file?

File: spec/PeridicBackgroundSync-index.bs (hunk 1 above)
Comment (mugdhalakhani) [reply spec/PeridicBackgroundSync-index.bs]: yes, I got a warning about them. I was planning to use them later which is why I'd left them here. I'll remove and re-add when I'm using them.

File: spec/PeridicBackgroundSync-index.bs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  <h2 id="constructs">Constructs</h2>
+  A <a>service worker registration</a> has an associated <dfn>list of periodicsync registrations</dfn> whose element type is a <a>periodicsync registration</a>.
+  
+  A <dfn>periodicsync registration</dfn> is a tuple consisting of a <a>tag</a>, a dictionary of <a>options</a>, and a <a lt="registration state">state</a>.
+
+  The <dfn>tag</dfn> is a DOMString. Within one <a>list of periodicsync registrations</a> each <a>periodicsync registration</a> MUST have a unique <a>tag</a>.
+
+  The <dfn>options</dfn> are a dictionary containing <a>minInterval</a>, a long long. <dfn>minInterval</dfn> is used to specify the minimum interval, in milliseconds, at which the periodic synchronization should happen.
+
+  The <dfn>registration state</dfn> is one of <dfn>pending</dfn>, <dfn>waiting</dfn>, <dfn>firing</dfn>, or <dfn>reregisteredWhileFiring</dfn>. It is initially set to <a>pending</a>.
```
Comment (rayankans) [inline spec/PeridicBackgroundSync-index.bs]: I know this is forked from the BackgroundSync spec, but will this state be necessary for PBS?

**Review by rayankans — COMMENTED** [review]
If you are going for a separate spec then why not start a new repo? Also since this is a new file let's use the newer bikeshed syntax ([=...=] instead of <a>...</a> and the such).

Otherwise mostly LG

File: spec/index.html (hunk 3 above)
Comment (mugdhalakhani) [reply spec/index.html]: done.

File: spec/PeridicBackgroundSync-index.bs (hunk 2 above)
Comment (mugdhalakhani) [reply spec/PeridicBackgroundSync-index.bs]: done.

File: spec/PeridicBackgroundSync-index.bs (hunk 1 above)
Comment (mugdhalakhani) [reply spec/PeridicBackgroundSync-index.bs]: Turns out I'm using a few of these -- removing them causes ambiguity in the definitions of some terms used in the text. I've removed all of the ones not currently being used.

File: spec/PeridicBackgroundSync-index.bs (hunk 4 above)
Comment (mugdhalakhani) [reply spec/PeridicBackgroundSync-index.bs]: Yes, some algorithms I'll add will need them.

**Comment by mugdhalakhani** [comment]
> If you are going for a separate spec then why not start a new repo? Also since this is a new file let's use the newer bikeshed syntax ([=...=] instead of ... and the such).
> 
> Otherwise mostly LG

Thanks Rayan! I've updated to the new syntax, PTAL.

The two specs are related, so keeping them in the same repo feels appropriate. It'll also get us exposure to anyone who is currently following this repo. I've created an Issue here to decide on where this spec text should live, please feel free to chime in there. I'll move the spec to wherever we agree upon on that thread:
https://github.com/WICG/BackgroundSync/issues/156

**Comment by mugdhalakhani** [comment]
Ping!

File: spec/PeridicBackgroundSync-index.bs [hunk 5]
```diff
@@ -0,0 +1,95 @@
+<pre class='metadata'>
+Title: Web Periodic Background Synchronization
+Status: CG-DRAFT
+ED: https://wicg.github.io/BackgroundSync/spec/
```
Comment (rayankans) [inline spec/PeridicBackgroundSync-index.bs]: https://wicg.github.io/BackgroundSync/spec/PeriodicBackgroundSync-index.html

File: spec/PeridicBackgroundSync-index.bs [hunk 6]
```diff
@@ -0,0 +1,95 @@
+<pre class='metadata'>
```
Comment (rayankans) [inline spec/PeridicBackgroundSync-index.bs]: Can you fix the file name? Peridic -> Periodic

File: spec/PeridicBackgroundSync-index.bs [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+  As this API relies on service workers, functionality provided by this API is only available in a [=secure context=].
+  ## Example ## {#example}
+  Requesting a [=periodic Background Sync opportunity=] at a daily frequency from a [=browsing context=]:
+
+  <pre class="lang-js">
+    async function getLatestNews() {
+      const registration = await navigator.serviceWorker.ready;
+      try {
+        registration.periodicSync.register('get-latest-news', {
```
Comment (rayankans) [inline spec/PeridicBackgroundSync-index.bs]: Add an await or this will never throw.

File: spec/PeridicBackgroundSync-index.bs [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+A [=service worker registration=] has an associated <dfn>list of periodicsync registrations</dfn> whose element type is a [=periodicsync registration=].
+
+A <dfn>periodicsync registration</dfn> is a tuple consisting of a [=tag=], a dictionary of [=options=], and a [=registration state=].
+
+The <dfn>tag</dfn> is a DOMString. Within one [=list of periodicsync registrations=] each [=periodicsync registration=] MUST have a unique [=tag=].
+
+The <dfn>options</dfn> are a dictionary containing [=minInterval=], a long long. <dfn>minInterval</dfn> is used to specify the minimum interval, in milliseconds, at which the periodic synchronization should happen.
+
+The <dfn>registration state</dfn> is one of <dfn>pending</dfn>, <dfn>waiting</dfn>, <dfn>firing</dfn>, or <dfn>reregisteredWhileFiring</dfn>. It is initially set to [=pending=].
```
Comment (rayankans) [inline spec/PeridicBackgroundSync-index.bs]: Add newline

File: spec/PeridicBackgroundSync-index.bs [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+Constructs {#constructs}
+=========================
+
+A [=service worker registration=] has an associated <dfn>list of periodicsync registrations</dfn> whose element type is a [=periodicsync registration=].
+
+A <dfn>periodicsync registration</dfn> is a tuple consisting of a [=tag=], a dictionary of [=options=], and a [=registration state=].
+
+The <dfn>tag</dfn> is a DOMString. Within one [=list of periodicsync registrations=] each [=periodicsync registration=] MUST have a unique [=tag=].
+
+The <dfn>options</dfn> are a dictionary containing [=minInterval=], a long long. <dfn>minInterval</dfn> is used to specify the minimum interval, in milliseconds, at which the periodic synchronization should happen.
```
Comment (rayankans) [inline spec/PeridicBackgroundSync-index.bs]: I'd just get rid of the options dictionary and use minInterval directly. It adds no value here, and there's no need to match the IDL type.

**Review by rayankans — APPROVED** [review]

File: spec/PeridicBackgroundSync-index.bs (hunk 9 above)
Comment (mugdhalakhani) [reply spec/PeridicBackgroundSync-index.bs]: We discussed this on another CL, but the actual API interface uses a dictionary type. This is to allow addition of more options in the future. While I don't think we should remove the dictionary from the API itself, I'll remove the definition of the dictionary and link directly to minInterval.

File: spec/PeridicBackgroundSync-index.bs (hunk 5 above)
Comment (mugdhalakhani) [reply spec/PeridicBackgroundSync-index.bs]: done.

File: spec/PeridicBackgroundSync-index.bs (hunk 6 above)
Comment (mugdhalakhani) [reply spec/PeridicBackgroundSync-index.bs]: done.

File: spec/PeridicBackgroundSync-index.bs (hunk 7 above)
Comment (mugdhalakhani) [reply spec/PeridicBackgroundSync-index.bs]: done.
